src/coreclr/jit/optimizer.cpp

```diff
@@ -8169,16 +8171,7 @@ bool Compiler::fgCreateLoopPreHeader(unsigned lnum)
     // Allocate a new basic block for the pre-header.
 
     const bool  isTopEntryLoop = loop.lpIsTopEntry();
-    BasicBlock* preHead;
-
-    if (isTopEntryLoop)
-    {
-        preHead = BasicBlock::bbNewBasicBlock(this, BBJ_NONE);
-    }
-    else
-    {
-        preHead = BasicBlock::bbNewBasicBlock(this, BBJ_ALWAYS, entry);
-    }
+    BasicBlock* preHead        = BasicBlock::New(this, BBJ_ALWAYS, (isTopEntryLoop ? top : entry));
```

Comment:
I think we can just always branch to entry now, the logic before was trying to optimize the case for top-entry loops and fall through, but we don't need to do that anymore.